### ISSUE TITLE
Use react-native-localize library when app launched first time to detect locale

### ISSUE THREAD

**Issue by sanjeevyadavIT** [salomaoluiz/BuyIt#106]
Hello, currently the **current locale** come from `AsyncStorage`, but when user launched app for the first time, and let say, user default language is english, the **current locale** which will be shown to him will be `ptBR`

App can use [`react-native-localize`](https://github.com/zoontek/react-native-localize) to get phone default current locale, and if locale value stored in `AsyncStorage` is null, that means user hasn't sent any preference and we can use phone default one.

[example code](https://github.com/alexakasanjeev/magento_react_native_graphql/blob/master/src/i18n/i18n.ts)

If you want I can work on this, else close the issue

**Comment by salomaoluiz** [comment]
Yeah, this has been in my radar, but the last weeks I don't have much time to work on this, if you are be able to add this feature, I be very grateful.

**Comment by alexakasanjeev** [comment]
Sure, you can assign me the issue, would like to work on it

**Comment by sanjeevyadavIT** [comment]
hello @salomaoluiz I'm working on it, but facing one issue,

## Problem
the way language tags are defined in `react-native-localize` library and detected is 

```js
"en-US","pt-BR"
```

the way you have defined in code is 

```ts
export type AppLocales = 'ptBR' | 'enUS';

i18n.translations = {
  ptBR: { ...ptBR },
  enUS: { ...enUS },
};
```

So when I am using [`findbestavailablelanguage()`](https://github.com/zoontek/react-native-localize#findbestavailablelanguage) it is not working

```ts
RNLocalize.findBestAvailableLanguage(Object.keys(i18n.translations)) // this is returning undefined
```

## Solution
but if change `i18n.translations` like this, then it is returning correct language

```ts

i18n.translations = {
  'pt-BR': { ...ptBR },
  'en-US': { ...enUS },
};

RNLocalize.findBestAvailableLanguage(Object.keys(i18n.translations)) // now it is working
```

should I modify "enUS" to "en-US" and "ptBR" to "pt-BR" everywhere including all test

**Comment by salomaoluiz** [comment]
Sure, no problem, you can change the variables.

**Comment by salomaoluiz** [comment]
Closed by #114
